src/csharp/Grpc.Auth/GoogleAuthInterceptors.cs

```diff
@@ -43,10 +43,28 @@ public static class GoogleAuthInterceptors
         /// <returns>The interceptor.</returns>
         public static AsyncAuthInterceptor FromCredential(ITokenAccess credential)
         {
+            if(credential is ITokenAccessWithExtraInformation credentialWithExtra)
```

Comment:
I think the overall design for ITokenAccessWithExtraInformation makes sense (not considering the details of the implementation).
The `credential is ITokenAccessWithExtraInformation credentialWithExtra` approach makes sense but the danger is that if one accidentally has an old version of Grpc.Auth (without these changes), the ITokenAccessWithExtraInformation object will be viewed as just a regular ITokenAccess and the extra headers will simply be ignored silently (and I can imagine users not knowing what's going on and filing issues that the new funtionality doesn't work). Have we considered that risk?

Reply:
I think this is the same as if one accidentally has an older version of Google.Apis.Auth, in which case the quota_project field in Json will be ignored when creating the token.
Either way, as you say, we will be silently ignoring the header. Server might or might not fail whether it was expecting the header for the credential or not (tbh I'm not sure about that). At worst, the operation will be billed on a different project than intended, but nonetheless a project that the token was authorized to bill.
I think this is something we can't control on the client library side. I will make sure to raise this internally.